workspace.spec

```diff
@@ -572,6 +572,378 @@ module Workspace {
 		string description;
 	} ProvenanceAction;
 
+	/* Credit Metadata-related objects and fields
+
+		Workspace objects have a list of zero or more sets of credit metadata (CM), or citation information for the object. This information allows data imported into KBase to retain the appropriate details for users wishing to cite the data source, and for those who contributed to its creation to be credited for their work.
+
+		CM is stored in a list as part of the object provenance, but unlike other provenance information, which is immutable, CM can be updated.
+
+		Credit metadata is added after object creation, and requires the user to be a workspace admin. Username, timestamp, and credit metadata schema version are stored along with the CM, which allows tracking of CM changes.
+
+		To update existing CM, a new set of credit metadata is added to the list.
+
+		When requesting credit metadata, the last (i.e. most recent) entry in the list is returned.
```

Comment:
I was under the impression that we would store the CM as a list, but only the most recent CM would be exposed to the user. In effect, the users don't see a list, they only see the most recent update. The list might be accessible via a specialized admin command or just direct db access for historical purposes. Is this wrong?

Reply:
No, that is correct -- I just haven't worded my description of it very well. I think it may be better to hide the implementation details from the spec file and just reflect the public API here, since both you and Elisha have had questions after reading that description!

Reply:
That makes sense to me. Any docs for admin functions that can access older entries would go here: https://ci.kbase.us/services/ws/docs/administrationinterface.html

Reply:
I  unresolved this as it looks like there's still references to the list, e.g. `Workspace objects have a list of zero or more sets of credit metadata (CM), or citation information for the object.`

Reply:
Removed that reference.

Reply:
:+1: